lib/bugsnag/middleware/suggestion_data.rb

```diff
@@ -0,0 +1,30 @@
+module Bugsnag::Middleware
+  class SuggestionData
+
+    CAPTURE_REGEX = /Did you mean\?([\s\S]+)$/
+    DELIMITER = "\n"
+
+    def initialize(bugsnag)
+      @bugsnag = bugsnag
+    end
+
+    def call(report)
+      matches = []
+      report.raw_exceptions.each do |exception| 
+        match = CAPTURE_REGEX.match(exception.message)
+        next unless match
+        
+        suggestions = match.captures[0].split(DELIMITER)
+        matches.concat suggestions.map{ |suggestion| suggestion.strip }
+      end
+
+      if matches.size == 1
+        report.add_tab(:"did you mean", {:suggestion => matches.first})
```

Comment:
Commonly in other notifiers we use an error tab i believe to generically have information about the error? Having a tab with one entry in it doesnt seem like something we would want to do long term

Reply:
That's a good idea. We could use "Error" as the tab name, in case there is other error-specific items to add later.

Reply:
As an idea I just had - its where the bugsnag_meta_data attached to the error could go?

Reply:
Would that risk making the 'custom' tab redundant?  It might be useful creating an 'error' tab and renaming the 'custom' tab to separate error generated vs user generated

Reply:
@Cawllec The custom tab is where metadata is collated if it does not conform to the expected format (e.g., not nested so that there is a tab name and a key name, like if you use `report.add_tab(:error, "foo")`). Its a bit out of scope of this changeset, but I think I see what you mean. Let's use `:error` as the tab name here though, and then we could separately use it as the tab for `bugsnag_meta_data` in a later feature.